libminifi/test/azure-tests/PutAzureDataLakeStorageTests.cpp

```diff
@@ -0,0 +1,271 @@
+
+const std::string FILESYSTEM_NAME = "testfilesystem";
+const std::string DIRECTORY_NAME = "testdir";
+const std::string FILE_NAME = "testfile.txt";
+const std::string CONNECTION_STRING = "test-connectionstring";
+const std::string TEST_DATA = "data123";
+const std::string GETFILE_FILE_NAME = "input_data.log";
+
+class MockDataLakeStorageClient : public minifi::azure::storage::DataLakeStorageClient {
+ public:
+  const std::string PRIMARY_URI = "http://test-uri/file";
+
+  bool createFile(const minifi::azure::storage::PutAzureDataLakeStorageParameters& /*params*/) override {
+    if (file_creation_error_) {
+      throw std::runtime_error("error");
+    }
+    return create_file_;
+  }
+
+  std::string uploadFile(const minifi::azure::storage::PutAzureDataLakeStorageParameters& params, const uint8_t* buffer, std::size_t buffer_size) override {
+    input_data_ = std::string(buffer, buffer + buffer_size);
+    params_ = params;
+
+    if (upload_fails_) {
+      throw std::runtime_error("error");
+    }
+
+    return RETURNED_PRIMARY_URI;
+  }
+
+  void setFileCreation(bool create_file) {
+    create_file_ = create_file;
+  }
+
+  void setFileCreationError(bool file_creation_error) {
+    file_creation_error_ = file_creation_error;
+  }
+
+  void setUploadFailure(bool upload_fails) {
+    upload_fails_ = upload_fails;
+  }
+
+  minifi::azure::storage::PutAzureDataLakeStorageParameters getPassedParams() const {
+    return params_;
+  }
+
+ private:
+  const std::string RETURNED_PRIMARY_URI = "http://test-uri/file?secret-sas";
+  bool create_file_ = true;
+  bool file_creation_error_ = false;
+  bool upload_fails_ = false;
+  std::string input_data_;
+  minifi::azure::storage::PutAzureDataLakeStorageParameters params_;
+};
+
+class PutAzureDataLakeStorageTestsFixture {
+ public:
+  PutAzureDataLakeStorageTestsFixture() {
+    LogTestController::getInstance().setDebug<TestPlan>();
+    LogTestController::getInstance().setDebug<minifi::core::Processor>();
+    LogTestController::getInstance().setTrace<minifi::core::ProcessSession>();
+    LogTestController::getInstance().setTrace<processors::GetFile>();
+    LogTestController::getInstance().setTrace<processors::PutFile>();
+    LogTestController::getInstance().setDebug<processors::UpdateAttribute>();
+    LogTestController::getInstance().setDebug<processors::LogAttribute>();
+    LogTestController::getInstance().setTrace<minifi::azure::processors::PutAzureDataLakeStorage>();
+
+    // Build MiNiFi processing graph
+    plan_ = test_controller_.createPlan();
+    auto mock_data_lake_storage_client = std::make_unique<MockDataLakeStorageClient>();
+    mock_data_lake_storage_client_ptr_ = mock_data_lake_storage_client.get();
+    put_azure_data_lake_storage_ = std::shared_ptr<minifi::azure::processors::PutAzureDataLakeStorage>(
+      new minifi::azure::processors::PutAzureDataLakeStorage("PutAzureDataLakeStorage", utils::Identifier(), std::move(mock_data_lake_storage_client)));
```

Comment:
Use `std::make_shared`

Reply:
`std::make_shared` does not work in this case as the constructor is private and `std::make_shared` does not have permission to call it.